packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1044 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+        myName: 'Quint',
+        notMyName: 'Sergio'
+      });
+    }, 'You cannot specify both a positional param (at position 0) and the hash argument `name`.');
+  }
+
+  ['@test yield to inverse']() {
+    this.registerComponent('my-if', {
+      template: strip`
+        {{#if predicate}}
+          Yes:{{yield someValue}}
+        {{else}}
+          No:{{yield to="inverse"}}
+        {{/if}}`
+    });
+
+    this.render(strip`
+      {{#my-if predicate=activated someValue=42 as |result|}}
+        Hello{{result}}
+      {{else}}
+        Goodbye
+      {{/my-if}}`,
+      {
+        activated: true
+      });
+
+    this.assertText('Yes:Hello42');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Yes:Hello42');
+
+    this.runTask(() => this.context.set('activated', false));
+
+    this.assertText('No:Goodbye');
+
+    this.runTask(() => this.context.set('activated', true));
+
+    this.assertText('Yes:Hello42');
+  }
+
+  ['@htmlbars parameterized hasBlock inverse'](assert) {
+    this.registerComponent('check-inverse', {
+      template: strip`
+        {{#if (hasBlock "inverse")}}
+          Yes
+        {{else}}
+          No
+        {{/if}}`
+    });
+
+    this.render(strip`
+      {{#check-inverse id="expect-no"}}{{/check-inverse}}
+      {{#check-inverse id="expect-yes"}}{{else}}{{/check-inverse}}`);
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+  }
+
+  ['@htmlbars parameterized hasBlock default'](assert) {
+    this.registerComponent('check-block', {
+      template: strip`
+        {{#if (hasBlock)}}
+          Yes
+        {{else}}
+          No
+        {{/if}}`
+    });
+
+    this.render(strip`
+      {{check-block id="expect-no"}}
+      {{#check-block id="expect-yes"}}{{/check-block}}`);
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+  }
+
+  ['@htmlbars non-expression hasBlock'](assert) {
+    this.registerComponent('check-block', {
+      template: strip`
+        {{#if hasBlock}}
+          Yes
+        {{else}}
+          No
+        {{/if}}`
+    });
+
+    this.render(strip`
+      {{check-block id="expect-no"}}
+      {{#check-block id="expect-yes"}}{{/check-block}}`);
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#expect-no').text(), 'No');
+    assert.equal(this.$('#expect-yes').text(), 'Yes');
+  }
+
+  ['@htmlbars parameterized hasBlockParams'](assert) {
```

Comment:
I think the test was incorrectly named – I believe "parameterized" refers to passing an "argument" to the helper, i.e. `(hasBlockParams "foo")`, which is not a thing afaik. So I think we can just say "expression hasBlockParams" here.